### PR TITLE
DO NOT MERGE Add tests for getUserMedia constraints

### PR DESCRIPTION
**Pull request by elhesuu**
This is the setup used for GUM constraints tests. I'm opening this PR mostly for future reference, and maybe in case this can help at least partially automating device testing.

On one hand, on `/tests/constraints/index.html` it will prompt for permission and then call `getUserMedia` with array of constraints. Results are then output in markdown-ish format (on a single line) to allow copying from Browserstack Live (multiple lines or nested block elements did not allow so).

On the other hand I tried automating these tests using Selenium and Browserstack Automate, but results are not promising, since for getting access to the camera with Chrome `--use-fake-ui-for-media-stream` and `--use-fake-device-for-media-stream` flags need to be set, and results are exactly the same for all devices. Using `--use-fake-ui-for-media-stream` does not work, perhaps because the browser cannot access the real device camera.

Results can be seen here
https://docs.google.com/spreadsheets/d/1V_0Q-ZMGUZ4P2J8N-6aQEqflLbpfNwquv1Do9kxn8UQ/edit?usp=sharing

### CODE REVIEW

**Comment by rfreitas** [comment]
is the goal to merge this?

**Comment by rfreitas** [comment]
any conclusions on the best combination?

**Comment by elhesuu** [comment]
Hey @rfreitas, didn't intend to merge this, but added you as reviewer just in case you hadn't already seen this.

Based on the results safe combination should be:

```
{
  video: {
    width: 1280,
    height: 720, 
    facingMode: 'user',
  }
}
```

A good fallback should be:

```
{
  video: {
    height: 480,
    facingMode: 'user',
  }
}
```

**Comment by rfreitas** [comment]
so have we confirmed that `video: { height: 480, facingMode: 'user'}` is better than `video:{facingMode: 'user'}` ?

**Comment by rfreitas** [comment]
Also, it doesn't seem that we tested if `video: { width: 1280, height: 720}` is better than `video: {height: 720}`

My intuition is still that less constraints is safer :/

**Comment by elhesuu** [comment]
Of all devices I tested on `video: { width: 1280, height: 720}` didn't fail once. 
It should be noted that my tests were made on Android devices mostly, and my own Macbook pro, no iOS or windows devices were used. So your intuition might be right.

So I think it would be good to test this on the highest number of real devices possible. If so, I think this setup could be of help.

File: test/constraints/index.html [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        console.log(err);
+        this.setState({
+          hasPermission: !!result,
+          error: err.name || err.message,
+        });
+      }
+
+      renderTests() {
+        return [
+          ...sizes.map(size => ({ ...size, facingMode: 'user' })),
```
[the remark this answers is not in the document]
Comment (elhesuu) [reply test/constraints/index.html]: mapping and returning a new array of objects? Should I try to refactor?

**Comment by seewah** [comment]
Can we at least test this on iOS? is there a link I can use to test? I can test on the windows computer over here too

**Pull request closed without merging**